File: backend/app/models/agent.py
"""Pydantic models for ITOM agent representation.

Defines the data structures used by the GET /api/agents endpoint to describe
available ITOM agents and their operational status.
"""

from enum import Enum

from pydantic import BaseModel, Field


class AgentStatus(str, Enum):
    """Operational status of an ITOM agent."""

    ONLINE = "online"
    OFFLINE = "offline"
    BUSY = "busy"


class AgentDomain(str, Enum):
    """Functional domain of an ITOM agent."""

    DISCOVERY = "discovery"
    ASSET = "asset"
    AUDIT = "audit"
    DOCUMENTATION = "documentation"
    ORCHESTRATOR = "orchestrator"


class AgentResponse(BaseModel):
    """Response model for a single ITOM agent.

    Returned as part of the GET /api/agents list and used by the frontend
    AgentSelector component to display agent options with live status.
    """

    id: str = Field(description="Unique identifier for the agent (e.g., 'discovery', 'asset').")
    name: str = Field(description="Human-readable display name (e.g., 'Discovery Agent').")
    description: str = Field(description="Short description of the agent's purpose.")
    status: AgentStatus = Field(description="Current operational status.")
    domain: AgentDomain = Field(description="The ITOM domain this agent operates in.")
    icon: str | None = Field(
        default=None,
        description="Optional icon identifier for display in the UI.",
    )

File: frontend/src/types/agent.ts
/**
 * Operational status of an ITOM agent.
 * - "online"  -- Agent is running and accepting requests
 * - "offline" -- Agent is not available
 * - "busy"    -- Agent is processing a request and may have increased latency
 */
export type AgentStatus = "online" | "offline" | "busy";

/**
 * The functional domain of an ITOM agent.
 */
export type AgentDomain =
  | "discovery"
  | "asset"
  | "audit"
  | "documentation"
  | "orchestrator";

/**
 * An ITOM agent that can participate in conversations.
 *
 * Agents are specialized services that handle specific ITOM domains.
 * The frontend fetches agent information from the backend, which queries
 * the orchestrator for real-time status.
 */
export interface Agent {
  /** Unique identifier for the agent (e.g., "discovery", "asset", "auditor"). */
  id: string;

  /** Human-readable display name (e.g., "Discovery Agent"). */
  name: string;

  /** Short description of what this agent does. */
  description: string;

  /** Current operational status. */
  status: AgentStatus;

  /** The ITOM domain this agent operates in. */
  domain: AgentDomain;

  /** Optional icon identifier or URL for display in the UI. */
  icon?: string;
}
